install/preview/entrypoint.sh

```diff
@@ -0,0 +1,131 @@
+
+cat << EOF > /var/lib/rancher/k3s/server/manifests/ca-pair.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ca-key-pair
+data:
+  ca.crt: $(base64 -w0 "${HOME}"/.local/share/mkcert/rootCA.pem)
+  tls.crt: $(base64 -w0 "${HOME}"/.local/share/mkcert/rootCA.pem)
+  tls.key: $(base64 -w0 "${HOME}"/.local/share/mkcert/rootCA-key.pem)
+EOF
+
+cat << EOF > /var/lib/rancher/k3s/server/manifests/issuer.yaml
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: ca-issuer
+spec:
+  ca:
+    secretName: ca-key-pair
+EOF
+
+echo "creating Gitpod SSL secret..."
+cat << EOF > /var/lib/rancher/k3s/server/manifests/https-cert.yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: https-cert
+spec:
+  secretName: https-certificates
+  issuerRef:
+    name: ca-issuer
+    kind: Issuer
+  dnsNames:
+    - "$DOMAIN"
+    - "*.$DOMAIN"
+    - "*.ws.$DOMAIN"
+EOF
+
+mkdir -p /var/lib/rancher/k3s/server/manifests/gitpod
+
+/gitpod-installer init > config.yaml
+yq e -i '.domain = "'"${DOMAIN}"'"' config.yaml
+yq e -i '.certificate.name = "https-certificates"' config.yaml
+yq e -i '.certificate.kind = "secret"' config.yaml
+yq e -i '.customCACert.name = "ca-key-pair"' config.yaml
+yq e -i '.customCACert.kind = "secret"' config.yaml
+yq e -i '.observability.logLevel = "debug"' config.yaml
+yq e -i '.workspace.runtime.containerdSocket = "/run/k3s/containerd/containerd.sock"' config.yaml
+yq e -i '.workspace.runtime.containerdRuntimeDir = "/var/lib/rancher/k3s/agent/containerd/io.containerd.runtime.v2.task/k8s.io/"' config.yaml
+
+echo "extracting images to download ahead..."
+/gitpod-installer render --config config.yaml | grep 'image:' | sed 's/ *//g' | sed 's/image://g' | sed 's/\"//g' | sed 's/^-//g' | sort | uniq > /gitpod-images.txt
+echo "downloading images..."
+while read -r image "$(cat /gitpod-images.txt)"; do
+   # shellcheck disable=SC2154
+   ctr images pull "$image" >/dev/null &
+done
+
+ctr images pull "docker.io/gitpod/workspace-full:latest" >/dev/null &
+
+/gitpod-installer render --config config.yaml --output-split-files /var/lib/rancher/k3s/server/manifests/gitpod
+for f in /var/lib/rancher/k3s/server/manifests/gitpod/*.yaml; do (cat "$f"; echo) >> /var/lib/rancher/k3s/server/gitpod.debug; done
+rm /var/lib/rancher/k3s/server/manifests/gitpod/*NetworkPolicy*
+for f in /var/lib/rancher/k3s/server/manifests/gitpod/*PersistentVolumeClaim*.yaml; do yq e -i '.spec.storageClassName="local-path"' "$f"; done
+yq eval-all -i ". as \$item ireduce ({}; . *+ \$item)" /var/lib/rancher/k3s/server/manifests/gitpod/*_StatefulSet_messagebus.yaml /app/manifests/messagebus.yaml
+for f in /var/lib/rancher/k3s/server/manifests/gitpod/*StatefulSet*.yaml; do yq e -i '.spec.volumeClaimTemplates[0].spec.storageClassName="local-path"' "$f"; done
+
+# removing init container from ws-daemon (systemd and Ubuntu)
+yq eval-all -i 'del(.spec.template.spec.initContainers[0])' /var/lib/rancher/k3s/server/manifests/gitpod/*_DaemonSet_ws-daemon.yaml
```

Comment:
This a quite a lot of pre-processing. Could you add comments on why they are actually needed?

Reply:
Added them in https://github.com/gitpod-io/gitpod/pull/10695